### PR TITLE
Upgrade to ember-cli 2.17.1

### PR DESCRIPTION
**Pull request by scottkidder**

### CODE REVIEW

**Comment by jbescoyez** [comment]
👍  - I'd love to see this merged. It would remove an ugly deprecation I have on canary by updating `ember-cli-version-checker`. cc/ @poteto 

```
DEPRECATION: An addon is trying to access project.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider doing: require("resolve").sync(something, { basedir: project.root }). Accessed from:   DependencyVersionChecker.NPMDependencyVersionChecker ([...]/node_modules/ember-deep-set/node_modules/ember-cli-version-checker/src/npm-dependency-version-checker.js:11:32)
```

**Comment by scottkidder** [comment]
Anything I can do to get this mergede? I imagine people will start finding their way here with the same deprecation @jbescoyez has noticed now that 2.18.0 has been released.

File: package.json
Comment (poteto) [inline package.json]: Can you remove this package please?

**Review by poteto — CHANGES_REQUESTED** [review]

**Comment by poteto** [comment]
LGTM, just need to remove `ember-welcome-page`.

File: yarn.lock
Comment (nucleartide) [inline yarn.lock:2]: If you're removing `ember-welcome-page`, could you also update `yarn.lock`?

Comment (scottkidder) [reply yarn.lock:2]: Of course, don't know how I missed that.

**Comment by nucleartide** [comment]
Looks like Travis couldn't download Node, just restarted the build.

**Comment by scottkidder** [comment]
@nucleartide Is there something holding this back yet?

**Comment by nucleartide** [comment]
@scottkidder Could you verify that the deprecation message goes away by pinning `ember-deep-set` to this branch's commit hash?

**Review by poteto — APPROVED** [review]

**Comment by poteto** [comment]
LGTM

**Comment by scottkidder** [comment]
> @scottkidder Could you verify that the deprecation message goes away by pinning ember-deep-set to this branch's commit hash?

I have verified this.

**Comment by nucleartide** [comment]
Awesome, thanks! Going to merge this.

@poteto I don't have publish access :( would you mind version-bumping/publishing?

**Comment by poteto** [comment]
@nucleartide Just corrected that!

**Comment by jbescoyez** [comment]
@poteto @nucleartide Would you mind releasing a new npm version? 
So that we will be able to update the dependency in poteto/ember-changeset which is causing the deprecation message.

Otherwise, I just open a PR mentioning HEAD commit: https://github.com/poteto/ember-changeset/pull/279

Tnx :)

**Comment by nucleartide** [comment]
So I'm kinda iffy about this. As a maintainer of `ember-changeset` and its community of plugins, my goal is to get to a point where actual maintenance isn't necessary, because `ember-changeset` solves a problem so flawlessly. This addon is quite small, and I think having to do things like updating ember-cli and bumping versions and publishing shouldn't be necessary, or should be automated.

I guess the root of the problem is depending on a framework in the first place, but I know there is ongoing work on making Ember more npm-friendly. (For example, an alternative solution might be to consume `lodash.setwith` and provide an Ember-specific customizer.)

For those reading – do you see this as an urgent problem? I'd like to understand how people perceive the deprecation messages.

**Comment by scottkidder** [comment]
Agreed that this maintenance isn't ideal, but I think this is urgent and addressing deprecation messages is important right now as some of us are trying to get to 3.0

**Comment by jbescoyez** [comment]
@nucleartide I've always fought deprecation messages in my code. It is a sign that you are not following the new guidelines and that you'll get an error at some point. By fixing them as they appear, my code always stays in an healthy state.

From your perspective what would be the right time to merge that and update npm versions? Also, I'm not sure to understand what would be the alternative? Can I force `ember-changeset` to use a specific revision of `ember-deep-set`?

**Comment by nucleartide** [comment]
Yes, there isn't much of an alternative. Here are the necessary steps:

1. [x] Figure out why `ember-deep-set`'s Travis build is failing.
2. [x] Add ember-cli-release to ember-deep-set's dependencies.
3. [x] Publish `ember-deep-set`.
4. [x] Update the `ember-deep-set` version in `ember-changeset`.
5. [x] Version bump `ember-changeset`.
6. [x] Publish `ember-changeset`.
7. [x] Update the `ember-changeset` version in `ember-changeset-validations`.
8. [x] Version bump `ember-changeset-validations`.
9. [x] Publish `ember-changeset-validations`.

(There is probably a solution in https://github.com/ember-cli/rfcs/pull/112 re: peer dependencies, but I don't have time to look into it.)

However, I notice the Travis build for `ember-deep-set` is failing on master. It looks like some Travis config issue (see https://github.com/travis-ci/travis-ci/issues/9024).

@jbescoyez Would you mind looking into the Travis issue? We can go through the list of tasks together.

**Comment by jbescoyez** [comment]
@nucleartide I'm a bit late on this conversation. It looks like the Travis build is passing again (maybe you fixed it?). I'm ready to help if needed.

**Comment by nucleartide** [comment]
@jbescoyez Yep, the Travis build was fixed by https://github.com/poteto/ember-deep-set/pull/12.

Just published new versions of `ember-changeset` and `ember-changeset-validations` and tested that the deprecation message goes away – lemme know if there are any problems with using them!

**Comment by jbescoyez** [comment]
@nucleartide I confirm. No more deprecation warning! Many tnx.

**Pull request merged**
